diff --git a/src/evaluation/src/util/init_sta.hh b/src/evaluation/src/util/init_sta.hh
index 08cfcc4ab..10dcb1ff0 100644
--- a/src/evaluation/src/util/init_sta.hh
+++ b/src/evaluation/src/util/init_sta.hh
@@ -28,6 +28,7 @@
 #include <string>
 #include <unordered_map>
 #include <vector>
+#include <optional>
 
 namespace ista {
 enum class AnalysisMode;
diff --git a/src/vectorization/database/CMakeLists.txt b/src/vectorization/database/CMakeLists.txt
index 1552d9f7a..3f92e42bd 100644
--- a/src/vectorization/database/CMakeLists.txt
+++ b/src/vectorization/database/CMakeLists.txt
@@ -13,3 +13,4 @@ target_include_directories(ivec_db
     PUBLIC
     ${CMAKE_CURRENT_SOURCE_DIR}
 )
+target_compile_features(ivec_db PUBLIC cxx_std_20)
\ No newline at end of file
diff --git a/src/vectorization/database/vec_node.h b/src/vectorization/database/vec_node.h
index 19c40ac6f..03df0d5ec 100644
--- a/src/vectorization/database/vec_node.h
+++ b/src/vectorization/database/vec_node.h
@@ -30,6 +30,7 @@
 #include <set>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 namespace ivec {
 
diff --git a/src/vectorization/src/graph/data_manager/CMakeLists.txt b/src/vectorization/src/graph/data_manager/CMakeLists.txt
index 029dcfb18..9c2484243 100644
--- a/src/vectorization/src/graph/data_manager/CMakeLists.txt
+++ b/src/vectorization/src/graph/data_manager/CMakeLists.txt
@@ -1,6 +1,9 @@
 set(CMAKE_AUTOMOC ON)
 set(CMAKE_AUTORCC ON)
 set(CMAKE_AUTOUIC ON)
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
 option(BUILD_VEC_GUI "Enable GUI components (default OFF)" OFF)
 
 if(POLICY CMP0100)
diff --git a/src/vectorization/src/graph/data_manager/vec_net_graph_gen.cc b/src/vectorization/src/graph/data_manager/vec_net_graph_gen.cc
index 613ec1c5e..33a48ef91 100644
--- a/src/vectorization/src/graph/data_manager/vec_net_graph_gen.cc
+++ b/src/vectorization/src/graph/data_manager/vec_net_graph_gen.cc
@@ -23,6 +23,7 @@
  */
 
 #include <algorithm>
+#include <ranges>
 #include <boost/graph/connected_components.hpp>
 #include <boost/graph/dijkstra_shortest_paths.hpp>
 #include <boost/graph/kruskal_min_spanning_tree.hpp>
